src/contract/HandlerBasedContract.ts

```diff
@@ -798,14 +799,19 @@ export class HandlerBasedContract<State> implements Contract<State> {
     return this._parentContract == null;
   }
 
-  async getStorageValues(keys: string[]): Promise<Map<string, string | null>> {
+  async getStorageValues(keys: string[]): Promise<Map<string, StorageValue>> {
+    const lastSortKey = await this.warp.stateEvaluator.lastCachedSortKey();
     const storage = this.warp.kvStorageFactory(this.txId());
-    const result: Map<string, string> = new Map();
+    const result: Map<string, StorageValue> = new Map();
     try {
       await storage.open();
       for (const key of keys) {
-        const value = await storage.get(Buffer.from(key));
-        result.set(key, value == null ? null : value.toString());
+        const lastValue = await storage.getLessOrEqual(key, lastSortKey);
+        const valStorage = {
+          keyStorage: lastSortKey,
```

Comment:
that's not very effective...ie. if we ask for 100 values, we will return the same value (the same sortKey) 100 times.

The returned object structure should be:
```
{ 
  sortKey,
  result
}
```

Reply:
yep, this is way better, fixed